Source/utils/display.cpp

```diff
@@ -550,10 +645,21 @@ void ReinitializeIntegerScale()
 		ResizeWindow();
 		return;
 	}
+	if (renderer == nullptr) return;
 
-	if (renderer != nullptr && SDL_RenderSetIntegerScale(renderer, *GetOptions().Graphics.integerScaling ? SDL_TRUE : SDL_FALSE) < 0) {
+#ifdef USE_SDL3
+	int w, h;
+	SDL_RendererLogicalPresentation mode;
+	if (!SDL_GetRenderLogicalPresentation(renderer, &w, &h, &mode)) ErrSdl();
+	const SDL_RendererLogicalPresentation newMode = *GetOptions().Graphics.integerScaling
+	    ? SDL_LOGICAL_PRESENTATION_INTEGER_SCALE
+	    : SDL_LOGICAL_PRESENTATION_STRETCH;

```

Comment:
Stretch will not respect the aspect ratio in case we do not get the exact preferred size.
```suggestion
	    : SDL_LOGICAL_PRESENTATION_LETTERBOX;
```